### DISCUSSION TITLE
Parellel Suspense Fetching

### DISCUSSION THREAD

**Discussion by ConsoleTVs** [vercel/swr#1872] [Ideas]
Currently, if a component requires more than once fetch, the Suspense will first throw the first promise and after it's resolved, it'll throw the second, making it similar to:

```js
await getUser()
await getPosts()
```

```js
const { data: user } = useSWR("user", {
  suspense: true,
});

const { data: posts } = useSWR("posts", {
  suspense: true, 
});
```

I would like to suggest a way for us to determine all the fetching in parallel so that the behaviour is something like this:

```js
await Promise.all([getUser(), getPosts()])
```

Therefore, allowing us to throw a single promise in suspense mode and performing parallel fetching. This problem is more noticeable the more fetches you have.

<hr>

This is how I solved it in my `vswr` library (SWR for vue.js)

https://github.com/ConsoleTVs/vswr#parallel-suspense

**Comment by koba04** [comment]
https://github.com/vercel/swr/pull/168 is an RFC to solve this 👍

**Reply by ConsoleTVs** [reply]
Good to know!
